src/gui/addnewtorrentdialog.cpp

```diff
@@ -451,101 +451,100 @@ void AddNewTorrentDialog::browseButton_clicked()
 void AddNewTorrentDialog::renameSelectedFile()
 {
     const QModelIndexList selectedIndexes = ui->contentTreeView->selectionModel()->selectedRows(0);
-    if (selectedIndexes.size() != 1)
-        return;
-    const QModelIndex &index = selectedIndexes.first();
-    if (!index.isValid())
-        return;
+    if (selectedIndexes.size() != 1) return;
+
+    const QModelIndex modelIndex = selectedIndexes.first();
+    if (!modelIndex.isValid()) return;
+
     // Ask for new name
-    bool ok;
-    const QString new_name_last = AutoExpandableDialog::getText(this, tr("Rename the file"),
-                                                                tr("New name:"), QLineEdit::Normal,
-                                                                index.data().toString(), &ok).trimmed();
-    if (ok && !new_name_last.isEmpty()) {
-        if (!Utils::Fs::isValidFileSystemName(new_name_last)) {
-            MessageBoxRaised::warning(this, tr("The file could not be renamed"),
-                                      tr("This file name contains forbidden characters, please choose a different one."),
-                                      QMessageBox::Ok);
+    bool ok = false;
+    const QString newName = AutoExpandableDialog::getText(this, tr("Renaming"), tr("New name:"), QLineEdit::Normal, modelIndex.data().toString(), &ok)
+                            .trimmed();
+    if (!ok) return;
+
+    if (newName.isEmpty() || !Utils::Fs::isValidFileSystemName(newName)) {
+        MessageBoxRaised::warning(this, tr("Rename error"),
+                                  tr("The name is empty or contains forbidden characters, please choose a different one."),
+                                  QMessageBox::Ok);
+        return;
+    }
+
+    if (m_contentModel->itemType(modelIndex) == TorrentContentModelItem::FileType) {
+        // renaming a file
+        const int fileIndex = m_contentModel->getFileIndex(modelIndex);
+
+        const QString oldFileName = m_torrentInfo.fileName(fileIndex);
+        const QString oldFilePath = m_torrentInfo.filePath(fileIndex);
+        const QString newFileName = newName;
```

Comment:
can I keep it?
because I copied the code from other file and made a slight modify.
or use `const QString &newFileName`?